examples/sql_bank/main.go

```diff
@@ -49,76 +50,80 @@ func moveMoney(db *sql.DB) {
 			continue
 		}
 		amount := rand.Intn(*maxTransfer)
-		tx, err := db.Begin()
-		if err != nil {
-			log.Fatal(err)
-		}
-		startTime := time.Now()
-		query := fmt.Sprintf("SELECT id, balance FROM accounts WHERE id IN (%d, %d)", from, to)
-		rows, err := tx.Query(query)
-		elapsed := time.Now().Sub(startTime)
-		if elapsed > 10*time.Millisecond {
-			log.Infof("%s took %v", query, elapsed)
-		}
-		if err != nil {
-			if log.V(1) {
-				log.Warning(err)
-			}
-			if err = tx.Rollback(); err != nil {
-				log.Fatal(err)
+
+		switch *transferStyle {
+		case 0:
+			update := `
+UPDATE bank.accounts
+  SET balance = CASE id WHEN $1 THEN balance-$3 WHEN $2 THEN balance+$3 END
+  WHERE id IN ($1, $2)
+`
+			if _, err := db.Exec(update, from, to, amount); err != nil {
+				if log.V(1) {
+					log.Warning(err)
+				}
+				continue
 			}
-			continue
-		}
-		var fromBalance, toBalance int
-		for rows.Next() {
-			var id, balance int
-			if err = rows.Scan(&id, &balance); err != nil {
+			atomic.AddUint64(&numTransfers, 1)
+
+		case 1:
+			tx, err := db.Begin()
+			if err != nil {
 				log.Fatal(err)
 			}
-			switch id {
-			case from:
-				fromBalance = balance
-			case to:
-				toBalance = balance
-			default:
-				panic(fmt.Sprintf("got unexpected account %d", id))
-			}
-		}
-		startTime = time.Now()
-		update := fmt.Sprintf("UPDATE accounts SET balance=%d WHERE id=%d", fromBalance-amount, from)
-		if _, err = tx.Exec(update); err != nil {
-			if log.V(1) {
-				log.Warning(err)
+			query := fmt.Sprintf("SELECT id, balance FROM accounts WHERE id IN (%d, %d)", from, to)
```

Comment:
Not the fault of this change, but we should use placeholders instead of sprintf for all query construction.